app/models/client.rb

```diff
@@ -321,6 +321,52 @@ def target_id=(value)
     Doi.transfer(client_id: symbol.downcase, target_id: target.id)
   end
 
+  def transfer(options = {})
+
+    if options[:target_id].blank?
+      Rails.logger.error "[Transfer] No target provider provided."
+      return nil
+    end
+
+    target_provider = Provider.where(symbol: options[:target_id]).first
+
+    if target_provider.blank?
+      Rails.logger.error "[Transfer] Provider doesn't exist."
+      return nil
+    end
+    
+    if target_provider.member_type == "consortium"
+      Rails.logger.error "[Transfer] Consortiums cannot have repositories."
+      return nil
+    end
+
+    original_provider = Provider.where(symbol: provider_id).first
+    ## Transfer client
+    update_attribute(:allocator, target_provider.id)
+
+    # These prefixes are used by multiple clients
+    prefixes_to_keep = ["10.4124", "10.4225", "10.4226", "10.4227"]
+
+    # delete all associated prefixes
+    prefix_ids = original_provider.prefixes.reject{ |prefix| prefixes_to_keep.include?(prefix)}.pluck(:id)
+    prefixes = original_provider.prefix_ids.reject{ |prefix| prefixes_to_keep.include?(prefix)}
+
+    if prefix_ids.present?
+      response = ProviderPrefix.where("prefix_id IN (?)", prefix_ids).destroy_all
+      puts "#{response.count} provider prefixes deleted."
+    end
+
+    # Transfer DOIs
```

Comment:
Technically speaking we are not transferring DOIs. Can we agree on a common language?